### ISSUE TITLE
ez_sql_sqlite3.php Creating default object from empty value

### ISSUE THREAD

**Issue by dialysisreport** [ezSQL/ezSQL#52]
get_var, and get_row are generating this error using sqlite3 (PHP 5.4.19)

Warning: Creating default object from empty value in ez_sql_sqlite3.php on line 178

**Comment by jwheat** [comment]
I don't mean to yank this issue back open almost a year later, but I noticed the same thing happening when the server admin upgraded PHP here as well.  

I started seeing 

**PHP Warning:  Creating default object from empty value in ez_sql_oracle8_9.php on line xxx**

all throughout the error log.

We're using Oracle, so I'm getting that error in 2 places in ez_sql_oracle_8_9.php

Depending on your PHP config, I believe if you have strict enabled, you see a lot more than usual, and it did not like the fact that the objects were being created the way they are.  Granted they are just warnings, but that log gets pretty big pretty fast.

Here's what I discovered and how I fixed it.

Around line 235  these three lines were throwing the message

~~~
$this->col_info[($i-1)]->name = @OCIColumnName($stmt,$i);
$this->col_info[($i-1)]->type = @OCIColumnType($stmt,$i);
$this->col_info[($i-1)]->size = @OCIColumnSize($stmt,$i);
~~~

So I added this if statement right before that block of 3 lines
~~~
if ( !is_object($this->col_info) ) {
  $this->col_info[] = new stdClass;
}
~~~

and similarly around line 256 (its hard to tell now that I've added code) this line was tossing an error as well 

~~~
$this->last_result[$row_num]->{$col_title} = $col_content;
~~~
 
and added this right before it
~~~
if (!is_object($this->last_result)) {
   $this->last_result[] = new stdClass;
}	
~~~

Basically all its doing it checking to see if the object exists, and if not, create it before it tries to start making assignments.  

The errors have gone away, and everything operates as expected now.

Hope this helps someone that may be having the same issue.

-Jon

**Comment by gnowxilef** [comment]
could you create a PR for this?

**Comment by jwheat** [comment]
Sure thing, I just did

**Comment by vic32** [comment]
jwheat, I've been suffering the same warning issue for months at the library ez_sql_postgresql (warning in line 240) and with your help now it's fixed.
Many many thanks

**Comment by liborse** [comment]
Hi, I had similar problem on ez_sql_postgresql.php, many thanks to jwheat. Can you upgrade this file in github? I am here totally new...

So I had this error: PHP Warning:  Creating default object from empty value in /var/www/***/web/admin/lib/ez_sql_postgresql.php on line 242

So I added before "$this->col_info[$i]->name = pg_field_name($this->result,$i);" this from jwheat:

if ( !is_object($this->col_info) ) { 
      $this->col_info[] = new stdClass;
}

**Comment by LobsterMan** [comment]
I am using ezsql using composer. I can't add the fix to the code itself because the library gets pulled automatically using composer on deployment. Can this fix be added?

**Comment by jv2222** [comment]
Yep, add a pull request and we'll merge it.
